pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -166,13 +169,107 @@ type nodeHealthData struct {
 	lease                    *coordv1beta1.Lease
 }
 
+func (n *nodeHealthData) deepCopy() *nodeHealthData {
+	if n == nil {
+		return nil
+	}
+	return &nodeHealthData{
+		probeTimestamp:           n.probeTimestamp,
+		readyTransitionTimestamp: n.readyTransitionTimestamp,
+		status:                   n.status.DeepCopy(),
+		lease:                    n.lease.DeepCopy(),
+	}
+}
+
+type nodeHealthMap struct {
+	lock        sync.RWMutex
+	nodeHealths map[string]*nodeHealthData
+}
+
+func newNodeHealthMap() *nodeHealthMap {
+	return &nodeHealthMap{
+		nodeHealths: make(map[string]*nodeHealthData),
+	}
+}
+
+// getDeepCopy - returns copy of node health data.
+// It prevents data being changed after retrieving it from the map.
+func (n *nodeHealthMap) getDeepCopy(name string) *nodeHealthData {
+	n.lock.RLock()
+	defer n.lock.RUnlock()
+	return n.nodeHealths[name].deepCopy()
+}
+
+func (n *nodeHealthMap) set(name string, data *nodeHealthData) {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	n.nodeHealths[name] = data
+}
+
+type podUpdateItem struct {
+	namespace string
+	name      string
+}
+
+type evictionStatus int
+
+const (
+	unmarked = iota
+	toBeEvicted
+	evicted
+)
+
+// nodeEvictionMap stores evictionStatus data for each node.
+type nodeEvictionMap struct {
+	lock          sync.Mutex
+	nodeEvictions map[string]evictionStatus
+}
+
+func newNodeEvictionMap() *nodeEvictionMap {
+	return &nodeEvictionMap{
+		nodeEvictions: make(map[string]evictionStatus),
+	}
+}
+
+func (n *nodeEvictionMap) registerNode(nodeName string) {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	n.nodeEvictions[nodeName] = unmarked
+}
+
+func (n *nodeEvictionMap) unregisterNode(nodeName string) {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	delete(n.nodeEvictions, nodeName)
+}
+
+func (n *nodeEvictionMap) setStatus(nodeName string, status evictionStatus) bool {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	if _, exists := n.nodeEvictions[nodeName]; !exists {
+		return false
+	}
+	n.nodeEvictions[nodeName] = status
+	return true
+}
+
+func (n *nodeEvictionMap) getStatus(nodeName string) (evictionStatus, bool) {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+	if _, exists := n.nodeEvictions[nodeName]; !exists {
+		return unmarked, false
+	}
+	return n.nodeEvictions[nodeName], true
+}
+
 // Controller is the controller that manages node's life cycle.
 type Controller struct {
 	taintManager            *scheduler.NoExecuteTaintManager
 	podInformersSynced      cache.InformerSynced
 	tenantPartitionManagers []*nodeutil.TenantPartitionManager
 	tenantPartitionClients  []clientset.Interface
 	resourcePartitionClient clientset.Interface
+	podLister         corelisters.PodLister
```

Comment:
Shouldn't this be an array of corelisters.PodLister or just use PodLister from tenantPartitionManagers?

Reply:
It looks like commented out later. Can this line be removed?

Reply:
removed.